plugin-annotation/src/main/java/com/mapbox/maps/plugin/annotation/AnnotationManagerImpl.kt

```diff
@@ -105,29 +117,42 @@ abstract class AnnotationManagerImpl<G : Geometry, T : Annotation<G>, S : Annota
    *
    * @return the GeoJsonSource created
    */
-  abstract fun createSource(): GeoJsonSource
+  protected abstract fun createSource(): GeoJsonSource
 
   /**
    * Create the layer for managed annotations
    *
    * @return the layer created
    */
-  abstract fun createLayer(): L
+  protected abstract fun createLayer(): L
 
   /**
    * Set filter on the managed annotations.
    */
   abstract var layerFilter: Expression?
 
   protected fun initLayerAndSource() {
-    initializeDataDrivenPropertyMap()
-    source = createSource()
-    layer = createLayer()
-    if (belowLayerId == null) {
-      style.addLayer(layer)
-    } else {
-      style.addLayerBelow(layer, belowLayerId)
+    if (layer == null || source == null) {
```

Comment:
Just to confirm - we recreate `source` even it was __not__ `null` but `layer == null`. Is it expected behavior?

Reply:
layer and source are created at the same time, here check them all just a double confirm.